src/reports/components/report-sections/link-script-provider.tsx

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+export const addEventListenerForLink = (doc: Document) => {
+    const targetPageLink = doc.getElementById('target-page-link');
+
+    targetPageLink.addEventListener('click', event => {
+        const result = confirm(
+            'Are you sure you want to navigate away from the Accessibility Insights report?\n' +
+                'This link will open the target page in a new tab.\n\nPress OK to continue or ' +
+                'Cancel to stay on the current page.',
+        );
+
+        if (result === false) {
+            event.preventDefault();
+        }
+    });
+};
+
+export const getAddListenerForLink = (code: string | Function): string =>
```

Comment:
Besides the name, this is the same function we already have on 'src\reports\components\report-sections\collapsible-script-provider.tsx'. Should be good to extract this to it's own file and reuse on both places.

Reply:
On a second though, we may need a slightly different function to consider how we pass the reference to the `confirm` function.

Reply:
Changed the function slightly to include a reference to `confirm`